### PR TITLE
Add ETag and Cache-Control support

### PR DESCRIPTION
**Pull request by kevmoo**
- Add `generateETag` and `maxAge` parameters to `createStaticHandler` and `createFileHandler`
- Support `If-None-Match` header to return 304 Not Modified when ETags match
- Default to weak ETags based on file size and modification time
- Allow setting `Cache-Control` public max-age via the `maxAge` parameter
- Add comprehensive tests for ETags, If-None-Match, and Cache-Control logic

### CODE REVIEW

File: pkgs/shelf_static/lib/src/static_handler.dart [hunk 1]
```diff
@@ -39,12 +39,23 @@ final _defaultMimeTypeResolver = MimeTypeResolver();
...
     {bool serveFilesOutsidePath = false,
     String? defaultDocument,
     bool listDirectories = false,
     bool useHeaderBytesForContentType = false,
-    MimeTypeResolver? contentTypeResolver}) {
+    MimeTypeResolver? contentTypeResolver,
+    FutureOr<String?> Function(File, FileStat)? generateETag,
+    Duration? maxAge}) {
   final rootDir = Directory(fileSystemPath);
   if (!rootDir.existsSync()) {
```
Comment (brianquinlan) [inline pkgs/shelf_static/lib/src/static_handler.dart]: This is fine but FYI, we are doing 2 stat calls when only one is required ;-)

**Review by brianquinlan — APPROVED** [review]
